cpp/ql/src/experimental/Security/CWE/CWE-193/ConstantSizeArrayOffByOne.ql

```diff
@@ -109,22 +110,39 @@ predicate isConstantSizeOverflowSource(Field f, PointerAddInstruction pai, int d
 
 module PointerArithmeticToDerefConfig implements DataFlow::ConfigSig {
   predicate isSource(DataFlow::Node source) {
-    isConstantSizeOverflowSource(_, source.asInstruction(), _)
+    isConstantSizeOverflowSource(_, _, source.asInstruction(), _)
   }
 
   pragma[inline]
   predicate isSink(DataFlow::Node sink) { isInvalidPointerDerefSink1(sink, _, _) }
 }
 
+module MergedPathGraph = DataFlow::MergePathGraph<PointerArithmeticToDerefFlow::PathNode, FieldAddressToPointerArithmeticFlow::PathNode, PointerArithmeticToDerefFlow::PathGraph, FieldAddressToPointerArithmeticFlow::PathGraph>;
+class PathNode = MergedPathGraph::PathNode;
+module StitchedPathGraph implements DataFlow::PathGraphSig<PathNode>{
+  query predicate edges(PathNode a, PathNode b) {
+    MergedPathGraph::PathGraph::edges(a, b)
+    or
+    a.asPathNode2().getNode().(DataFlow::InstructionNode).asInstruction() = b.asPathNode1().getNode().(DataFlow::InstructionNode).asInstruction().(PointerAddInstruction).getLeft()
+  }
+
+  query predicate nodes(PathNode n, string key, string val) {
+    MergedPathGraph::PathGraph::nodes(n, key, val)
+  }
+
+  query predicate subpaths(PathNode arg, PathNode par, PathNode ret, PathNode out) {
+    MergedPathGraph::PathGraph::subpaths(arg, par, ret, out)
+  }
+}
 module PointerArithmeticToDerefFlow = DataFlow::Global<PointerArithmeticToDerefConfig>;
 
 from
-  Field f, PointerArithmeticToDerefFlow::PathNode source,
-  PointerArithmeticToDerefFlow::PathNode sink, Instruction deref, string operation, int delta
+  Field f, PathNode fieldSource, PathNode paiNode,
+  PathNode sink, Instruction deref, string operation, int delta
 where
-  PointerArithmeticToDerefFlow::flowPath(source, sink) and
+  PointerArithmeticToDerefFlow::flowPath(paiNode.asPathNode1(), sink.asPathNode1()) and
   isInvalidPointerDerefSink2(sink.getNode(), deref, operation) and
-  isConstantSizeOverflowSource(f, source.getNode().asInstruction(), delta)
-select source, source, sink,
+  isConstantSizeOverflowSource(f, fieldSource.asPathNode2(), paiNode.getNode().asInstruction(), delta)
+select paiNode, fieldSource, sink,
```

Comment:
We [usually](https://codeql.github.com/docs/writing-codeql-queries/creating-path-queries/#constructing-a-path-query) use the `DataFlow::Node` (instead of the `DataFlow::PathNode)` as the first column, right?
```suggestion
select paiNode.getNode(), fieldSource, sink,
```

Reply:
This is still unresolved @rdmarsh2

Reply:
I changed the `from` clause - `pai` is now the `PointerArithmeticInstruction` that was kept in the flow state.